### PR TITLE
Rnbeads integration and LMC intepreation functions

### PR DESCRIPTION
**Pull request by schmic05**
This branch contains mainly three functions: associating LMC proportions with sample traits, GO enrichment analysis (GOstats package) for LMC-specific CpGs and LOLA enrichment for the same CpGs. The functions are ready to be called from within FactorViz and I'd thus like to have them in the master branch. Please have a look at the code and accept the pull request, if you are convinced.

### CODE REVIEW

**Review by lutsik — CHANGES_REQUESTED** [review]
Should we stick to a the scheme first used in RnBeads, i.e. first merging branches into devel and then merging into master as new (sub)releases? What do you think?

File: R/contribution_interpretation.R [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+#' @details Each element in the returned list is of length \code{K}, displaying the p-value of the statistical assocation of the
+#'           contributions of the corresponding LMC to the sample grouping. The p-values are produced by comparing the LMC 
+#'           contributions in all sample comparisons defined by \code{\link{rnb.sample.groups}} on \code{rnb.set}. The employed test
+#'           statistic for pariwise comparison can be specified by \code{test.fun}, for groups defining more than one group
+#'           \code{\link{kruskal.test}} is employed.
+#' @author Michael Scherer
+#' 
+#' @noRd
+
+link.to.traits <- function(medecom.set,cg_subset,K,lambda,rnb.set,test.fun=t.test){
```
Comment (lutsik) [inline R/contribution_interpretation.R]: Should we create more generic functions for this, which use phenotype data frames as input? RnBeads wrappers can be added on top?

**Comment by schmic05** [comment]
I like this idea! But then it will be entirely your resonsibility to merge the devel branch into the master branch as soon as you think it's worth merging.

File: R/contribution_interpretation.R [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+#' This routine performs a statistical test to determine if the difference in LMC contributions is different for the sample groups
+#' defined in \code{rnb.set}.
+#' 
+#' @param medecom.set An object of type \code{\link{MeDeComSet}} as the result of \code{\link{runMeDeCom}} containing LMCs and their
+#'                     proportions in the samples. The Set can contain multiple runs for different values of K and lambda.
+#' @param cg_subset The subset of sites used for computing LMCs in \code{medecom.set}.
+#' @param K The K parameter, determining the number of LMCs to extract from \code{medecom.set}.
+#' @param lambda The lambda parameter, determining the regularization used for the LMCs in \code{medecom.set}.
+#' @param rnb.set An object of type \code{\link{RnBSet}} containing methylation data and metadata for the same samples for which 
+#'                 \code{medecom.set} was computed or a data.frame of sample annotations (ann_S)
```
Comment (schmic05) [inline R/contribution_interpretation.R:136]: This is the case, we can either input and RnBSet or a data.frame. Anyway, the parameter name is missleading, will change it.

**Pull request closed without merging**